marklogic-data-hub-central/ui/src/components/modeling/entity-type-modal/entity-type-modal.test.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import userEvent from "@testing-library/user-event";
+import EntityTypeModal from './entity-type-modal';
+
+describe('EntityTypeModal Component', () => {
+  test('Modal is not visible', () => {
+    const { queryByTestId } =  render(
+    <EntityTypeModal 
+      isVisible={false} 
+      toggleModal={jest.fn()}
+      newEntityAdded={jest.fn()}
+    />);
+
+    expect(queryByTestId('entity-type-modal')).toBeNull();
+  });
+
+  test('Valid Entity name is used', async () => {
+    const { getByTestId, getByText } =  render(
+      <EntityTypeModal 
+        isVisible={true} 
+        toggleModal={jest.fn()}
+        newEntityAdded={jest.fn()}
+      />);
+    expect(getByText(/Add Entity Type/i)).toBeInTheDocument();
+
+    await userEvent.type(getByTestId('name'), 'AnotherModel');
+    expect(getByTestId('name')).toHaveAttribute('value', 'AnotherModel');
+
+    await userEvent.type(getByTestId('description'), 'Testing');
+    expect(getByTestId('description')).toHaveAttribute('value', 'Testing');
+
+    userEvent.click(getByText('Add'));
+  });
+
+  test('Invalid Entity name shows error', async () => {
```

Comment:
Test is missing assertion for invalid entry.

Reply:
I've updated the test and design to disable the button if the name isn't valid. I've added an additional test that mocks 400 error response for duplicate entity names